src/prefect/engine/state.py

```diff
@@ -280,6 +280,9 @@ def __init__(
     ):
         super().__init__(message=message, result=result, cached_inputs=cached_inputs)
         self.start_time = pendulum.instance(start_time or pendulum.now("utc"))
+        run_count = prefect.context.get("task_run_count")
```

Comment:
This "feels" like a very fragile thing but I can't think of a better way to do it

Reply:
Yea that's fair; this field is populated in every task runner pipeline step (which is where all state transitions occur), but I understand your point.  I think the way to think about it is that this will be populated within a "run context" but not otherwise, and this key is always present within a run context.